app/src/org/commcare/logic/ArchivedFormRemoteRestore.java

```diff
@@ -25,11 +29,30 @@
     public static void pullArchivedFormsFromServer(String remoteUrl,
                                                    final FormRecordListActivity activity,
                                                    final CommCarePlatform platform) {
+
+        requestForms(activity, platform, remoteUrl, CommCareApplication.instance().getDataPullRequester(), false);
+    }
+
+    public static void pullArchivedFormsFromFile(String filePath,
+                                                 final FormRecordListActivity activity,
+                                                 final CommCarePlatform platform) {
+
+        File file = new File(filePath);
+        if (file != null && file.exists()) {
+            LocalFilePullResponseFactory.setRequestPayloads(new File[]{file});
+            requestForms(activity, platform, "fake-server-that-is-never-used", LocalFilePullResponseFactory.INSTANCE, true);
+        } else {
+            Toast.makeText(activity, "Restore file doesn't exist", Toast.LENGTH_LONG).show();
```

Comment:
Small, but message should say `"Payload file doesn't exist"` instead of `"Restore file"`